src/core/Akka/Configuration/Hocon/HoconObject.cs

```diff
@@ -228,6 +241,8 @@ internal HoconObject MergeImmutable(HoconObject other)
                         mergedValue.AppendValue(mergedObject);
                         thisItems[otherItem.Key] = mergedValue;
                     }
+                    else
+                        thisItems[otherItem.Key] = otherItem.Value;
```

Comment:
So for non-`HoconObject` types, we just ignored them on merge before?

Reply:
yeah, it causes a lot of missing object fields, I'm surprised that this bug went through the test suite

Reply:
Is there a simple test case we can add for this to the HOCON test suite? Thinking about it mostly as a regression test.

Reply:
We do have rigorous testing suite in the standalone HOCON project, I think we used to have a limited HOCON test suite in Akka, they might be accidentally removed.